tests/interpreters/test_tensor_interpreter.py

```diff
@@ -0,0 +1,28 @@
+from xdsl.dialects import tensor
+from xdsl.dialects.builtin import ModuleOp, TensorType, f32, i32
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.shaped_array import ShapedArray
+from xdsl.interpreters.tensor import TensorFunctions
+from xdsl.utils.test_value import TestSSAValue
+
+
+def test_tensor_empty():
+    interpreter = Interpreter(ModuleOp([]))
+    interpreter.register_implementations(TensorFunctions())
+    op = tensor.EmptyOp((), TensorType(f32, [3, 2]))
+    c = interpreter.run_op(op, ())
+    assert c[0] == ShapedArray([None], [3, 2])
```

Comment:
I'd expect this:
```suggestion
    assert c[0] == ShapedArray([0.0] * 6, [3, 2])
```

Reply:
This will be the case also when we migrate to having the ShapedArray be backed by bytes instead of a Python list with #2287